kernel/src/process.rs

```diff
@@ -1191,7 +1198,11 @@ impl<C: Chip> ProcessType for Process<'a, C> {
     fn debug_syscall_called(&self, last_syscall: Syscall) {
         self.debug.map(|debug| {
             debug.syscall_count += 1;
-            debug.last_syscall = Some(last_syscall);
+            debug.last_syscalls[4] = debug.last_syscalls[3];
+            debug.last_syscalls[3] = debug.last_syscalls[2];
+            debug.last_syscalls[2] = debug.last_syscalls[1];
+            debug.last_syscalls[1] = debug.last_syscalls[0];
+            debug.last_syscalls[0] = (Some(last_syscall), None);
```

Comment:
This is quite inefficient compared to a ring buffer. Especially if one wants to increase the history size (once it's configurable in `kernel/src/config.rs`).

We already have a ring buffer implementation in the kernel. The current ring buffer implementation may be non-trivial to use here (because it holds a mutable reference to the underlying buffer array). But if that's the case, I think we should rather fix the ring buffer API than inefficiently copying contents in memory.